NachoSpigot-Server/src/main/java/net/minecraft/server/EntityLiving.java

```diff
@@ -1789,7 +1790,34 @@ public void receive(Entity entity, int i) {
     }
 
     public boolean hasLineOfSight(Entity entity) {
-        return this.world.rayTrace(new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ), new Vec3D(entity.locX, entity.locY + (double) entity.getHeadHeight(), entity.locZ)) == null;
+        if (entity instanceof EntityPlayer && NachoConfig.enableImprovedHitReg) {
+            // Head height is 1,5725
+            // Split it into three to get a more accurate line of sight -> 0.52416667
+
+            double parts = entity.getHeadHeight() / 3;
+
+            boolean firstCondition = this.world.rayTrace(
+                    new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
+                    new Vec3D(entity.locX, entity.locY + (parts * 3), entity.locZ)
+            ) == null;
+
+            boolean secondCondition = this.world.rayTrace(
+                    new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
+                    new Vec3D(entity.locX, entity.locY + (parts * 2), entity.locZ)
+            ) == null;
+
+            boolean thirdCondition = this.world.rayTrace(
+                    new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
+                    new Vec3D(entity.locX, entity.locY + (parts * 1), entity.locZ)
+            ) == null;
+
+            return firstCondition | secondCondition | thirdCondition;
```

Comment:
it will execute all 3 checks even if one is already true so it will return true anyway (also, didn't you want to use `||`?) 
```suggestion
        return this.world.rayTrace(
                new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
                new Vec3D(entity.locX, entity.locY + (parts * 3), entity.locZ)
        ) == null || this.world.rayTrace(
                new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
                new Vec3D(entity.locX, entity.locY + (parts * 2), entity.locZ)
        ) == null || this.world.rayTrace(
                new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
                new Vec3D(entity.locX, entity.locY + (parts * 1), entity.locZ)
        ) == null;
```